qutip/measurement.py

```diff
@@ -169,3 +325,143 @@ def measure(op, state):
+            3. kets (transformed to projectors)
+    targets : list of ints, optional
+              Specifies a list of target "qubit" indices on which to apply
+              the measurement using qutip.qip.gates.expand_operator to expand
+              ops into full dimension.
+
+    Returns
+    -------
+    case : ops is a measurement observable (Qobj)
+    eigenvalues: List of floats
+        The list of eigenvalues of the measurement operator.
+    eigenstates_or_projectors: List of Qobj
+        If the state was a ket, return the eigenstates of the measurement
+        operator. Otherwise return the projectors onto the eigenstates.
+    probabilities: List of floats
+        The probability of measuring the state as being in the
+        corresponding eigenstate (and the measurement result being
+        the corresponding eigenvalue).
+    case : ops is a list of measurement operators (list of Qobjs)
+    collapsed_states : List of Qobjs
+                    the collapsed states obtained after measuring the qubits
+                    and obtaining the qubit specified by the target in the
+                    state specified by the index.
+    probabilities : List of floats
+                    the probability of measuring a state in a the state
+                    specified by the index.
+
+    """
+
+    if isinstance(ops, list):
+        return measurement_statistics_povm(state, ops)
+    else:
+        return measurement_statistics_observable(state, ops)
+
+
+def measure(state, ops, targets):
+    """
+    A dispatch method that provides measurement results
+    handling both observable style measurements
+    and projector style measurements(POVMs and PVMs).
+
+    Parameters
+    ----------
+    state : Qobj
+            The ket or density matrix specifying the state to measure.
+    ops : Qobj or list of Qobjs
+          - measurement observable
+          or
+          - list of measurement operators M_i or kets
+            Either:
+            1. specifying a POVM s.t. E_i = dagger(M_i) * M_i
+            2. projection operators if ops correspond to
+               projectors (s.t. E_i = dagger(M_i) = M_i)
+            3. kets (transformed to projectors)
+    targets : list of ints, optional
+              Specifies a list of target "qubit" indices on which to apply
+              the measurement using qutip.qip.gates.expand_operator to expand
+              ops into full dimension.
+
+    Returns
+    -------
+     case : ops is a measurement observable (Qobj)
+     measured_value : float
+         The result of the measurement (one of the eigenvalues of op).
+     state : Qobj
+         The new state (a ket if a ket was given, otherwise a density
+         matrix).
+     case : ops is a list of measurement operators (list of Qobjs)
+     index : float
+         The resultant index of the measurement.
+     state : Qobj
+         The new state (a ket if a ket was given, otherwise a density
+         matrix).
```

Comment:
Is this a good way to do it? @nathanshammah @ajgpitch @quantshah @hodgestar

Reply:
* please add a line descriptor under each variable, e.g., case (which is repeated twice on line 450 and 456).
* using :class:`qutip.Qobj` would make it clickable and take one to the API. Or even just :class:`.Qobj`.